parity/lib.rs

```diff
@@ -219,14 +220,18 @@ fn execute<Cr, Rr>(command: Execute, on_client_rq: Cr, on_updater_rq: Rr) -> Res
 /// binary.
 ///
 /// On error, returns what to print on stderr.
-pub fn start<Cr, Rr>(conf: Configuration, on_client_rq: Cr, on_updater_rq: Rr) -> Result<ExecutionAction, String>
+pub fn start<Cr, Rr>(
+	conf: Configuration,
+	logger: Arc<RotatingLogger>,
```

Comment:
This breaks the C-bindings in `parity-clib` you need to update them accordingly

Reply:
The difficult part of #8574 (and the reason why it wasn't done immediately when `parity-clib` was created) is that ideally the `RotatingLogger` wouldn't be exposed in the API.

However I guess we can ignore the problem in the name of "meh, who cares".

Reply:
We're not exposing `RotatingLogger` in `C API`, we expose only `Logger` params struct.

Reply:
I mentioned "when parity-clib was created", because the Rust lib and the C library were created at the same time.

The logging system is unfortunately a global variable, and thus it shouldn't be the job of a library to set it.
The consequence is that if a program wants to use parity-as-a-library, they cannot setup their own logging compatible with the `log` crate.
An ideal world, it is `main.rs` that would setup the logging and that's it. The C library would have an independent function (not tied to parity, just like `parity_set_panic_hook` is totally independent as well) that sets up the Rust logging and explicitly mentions in its documentation that this applies to the `log` Rust crate as a whole.